src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -1261,150 +1288,81 @@ def serialize_with_spans(
-            root_traces = self.visit_transactions(
-                roots,
-                transactions,
-                errors,
-                visited_transactions,
-                visited_errors,
-            )
+        with sentry_sdk.start_span(op="serialize", description="create parent map"):
+            parent_event_map = defaultdict(list)
+            serialized_transactions = []
+            for transaction in transactions:
+                parent_id = transaction["trace.parent_transaction"]
+                serialized_transaction = TraceEvent(
+                    transaction, parent_id, -1, span_serialized=True
+                )
+                if parent_id is None:
+                    if transaction["trace.parent_span"]:
+                        orphans.append(serialized_transaction)
+                    else:
+                        root_traces.append(serialized_transaction)
+                else:
+                    parent_event_map[parent_id].append(serialized_transaction)
+                serialized_transactions.append(serialized_transaction)
 
-        # At this point all the roots have their tree built. Remaining
-        # transactions are either orphan transactions or children of
-        # orphan transactions. Orphan transactions (unlike roots) have
-        # a parent_id but the parent_id wasn't found (dropped span).
-        # We get a sorted list of these transactions by start timestamp.
-        with sentry_sdk.start_span(op="serialize", description="orphans"):
-            remaining_transactions = self.calculate_remaining_transactions(
-                transactions, visited_transactions
-            )
+        parent_error_map = defaultdict(list)
+        for error in errors:
+            if "trace.transaction" in error:
+                parent_error_map[error["trace.transaction"]].append(self.serialize_error(error))
+            else:
+                orphan_errors.append(error)
 
-            # Determine orphan transactions. `trace.parent_transaction` on a
-            # transaction is set when the indexed spans dataset has a row for
-            # the parent span id for this transaction. Since we already considered
-            # the root spans cases, the remaining spans with no parent transaction
-            # id are orphan transactions.
-            orphan_roots = [
-                orphan
-                for orphan in remaining_transactions
-                if orphan["trace.parent_transaction"] is None
-            ]
-
-            # Build the trees for all the orphan transactions.
-            orphans = self.visit_transactions(
-                orphan_roots,
-                remaining_transactions,
-                errors,
-                visited_transactions,
-                visited_errors,
-            )
+        with sentry_sdk.start_span(op="serialize", description="associate children"):
+            visited_transactions: set[str] = {
```

Comment:
don't feel too strongly about this, you'd need to change the rest obv
```suggestion
            visited_transaction_ids: set[str] = {
```